### PR TITLE
Sample parallel loop body

### PR DESCRIPTION
**Pull request by LaurentBerger**
An example to use ParallelLoopBody

### CODE REVIEW

File: samples/cpp/ParallelLoopBody.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    };
+};
+
+
+bool verbose=true;
+
+
+const String keys =
+"{help h usage ? |      | This program demonstrated the use of class ParallelLoopBody for threading\n ./ParallelLoopBody [image_name -- default ../data/lena.jpg]  --thread <int> default 16}"
+"{@image         |f:/lib/opencv/samples//data/lena.jpg      | image to process     }"
```
Comment (alalek) [inline samples/cpp/ParallelLoopBody.cpp]: Please eliminate absolute path.

**Comment by vpisarev** [comment]
@LaurentBerger, thanks! For the file names we prefer to use "lowercase" naming style. Could you please rename the sample?

File: samples/cpp/ParallelLoopBody.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+using namespace cv;
+using namespace std;
+
+// This class use ParallelLoopBody to process image using thread
+// ParallelMeanLine process image to have for each line a mean value of 128 and standard deviation of 64
+class ParallelMeanLine: public cv::ParallelLoopBody
+{
+private:
+    cv::Mat &imgSrc;
```
Comment (vpisarev) [inline samples/cpp/ParallelLoopBody.cpp]: cv:: is not needed here

File: samples/cpp/ParallelLoopBody.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    };
+};
+
+
+
+
+const String keys =
+"{help h usage ? |      | This program demonstrated the use of class ParallelLoopBody for threading\n ./ParallelLoopBody [image_name -- default ../data/lena.jpg]  --thread <int> default 16}"
+"{@image         |../data/lena.jpg      | image to process     }"
+    "{@thread        |16    | number o fthread     }"
```
Comment (vpisarev) [inline samples/cpp/ParallelLoopBody.cpp]: looks like there is typo here, should be "number of threads"

File: samples/cpp/ParallelLoopBody.cpp [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    // We want : all lines must have a mean value of 128 and standard deviation of 64. Negative value or value greater than 255 are threshold
+        for (int y = range.start; y < range.end; y++)
+        {
+
+            uchar *ptr=imgSrc.ptr(y);
+            int v;
+            for (int x = 0; x<h; x++,ptr++)
+            {
+                   v = int((*ptr-meanLine[y])*sqrt(64*64/stdLine[y])+128);
+                   if (v>255)
```
Comment (vpisarev) [inline samples/cpp/ParallelLoopBody.cpp]: please, use saturate_cast<uchar>() here

File: samples/cpp/ParallelLoopBody.cpp [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        imgSrc(img),
+        meanLine(m),
+        stdLine(s),
+        verbose(false)
+    {}
+    void Verbose(bool b){verbose=b;}
+    virtual void operator()(const Range& range) const
+    {
+
+        int h = imgSrc.cols;
```
Comment (vpisarev) [inline samples/cpp/ParallelLoopBody.cpp]: h is usually used to denote image height; probably, it should be renamed to w (for width) or ncols or something like that

File: samples/cpp/ParallelLoopBody.cpp [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+#include <opencv2/imgproc/imgproc.hpp>
+#include "opencv2/highgui/highgui.hpp"
+#include <iostream>
+
+using namespace cv;
+using namespace std;
+
+// This class use ParallelLoopBody to process image using thread
+// ParallelMeanLine process image to have for each line a mean value of 128 and standard deviation of 64
+class ParallelMeanLine: public cv::ParallelLoopBody
```
Comment (vpisarev) [inline samples/cpp/ParallelLoopBody.cpp]: what's the purpose of such preprocessing? Can we come up with something more practical?

File: samples/cpp/ParallelLoopBody.cpp (hunk 2 above)
Comment (LaurentBerger) [inline samples/cpp/ParallelLoopBody.cpp]: OK

File: samples/cpp/ParallelLoopBody.cpp (hunk 3 above)
Comment (LaurentBerger) [inline samples/cpp/ParallelLoopBody.cpp]: OK

File: samples/cpp/ParallelLoopBody.cpp (hunk 4 above)
Comment (LaurentBerger) [inline samples/cpp/ParallelLoopBody.cpp]: OK

File: samples/cpp/ParallelLoopBody.cpp (hunk 5 above)
Comment (LaurentBerger) [inline samples/cpp/ParallelLoopBody.cpp]: OK

File: samples/cpp/ParallelLoopBody.cpp (hunk 6 above)
Comment (LaurentBerger) [inline samples/cpp/ParallelLoopBody.cpp]: My objective is to give an ParrallelLoopBody example with a simple calculus (mean and variance line) and without any call to opencv functiong using ParallelLoopBody or thread.

**Pull request closed without merging**